pkg/hubble/exporter/exporter.go

```diff
@@ -77,11 +77,11 @@ func NewExporter(logger *slog.Logger, options ...Option) (*exporter, error) {
 
 // newExporter let's you supply your own WriteCloser for tests.
 func newExporter(logger *slog.Logger, opts Options) (*exporter, error) {
-	writer, err := opts.NewWriterFunc()
```

Comment:
Nitpick:  IMO it's a bit weird that we have a method that returns a function that we always immediately call. Couldn't we directly call `newWriterFunc` when calling `NewWriterFunc`, instead of returning it.  i.e.

```go
func (o *Options) NewWriterFunc() (io.WriteCloser, error) {
	return o.newWriterFunc()
}
```

Same for `NewEncoderFunc`

Reply:
Yes I felt the same when I made the patch. I weighed it with the fact that the method name contains `Func` which at least conveys some meaning as to why we do this.

v1.17 patch was merged before I could see this comment, oups! We'll keep it in-sync but noted if we work again in this part of the code.